reactor-core/src/test/java/reactor/ReactorTestExecutionListener.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2020-Present Pivotal Software Inc, All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package reactor;
+
+import org.junit.platform.engine.TestExecutionResult;
+import org.junit.platform.launcher.TestExecutionListener;
+import org.junit.platform.launcher.TestIdentifier;
+import reactor.core.publisher.Hooks;
+import reactor.core.scheduler.Schedulers;
+
+public class ReactorTestExecutionListener implements TestExecutionListener {
+
+	public static void reset() {
+		Hooks.resetOnOperatorDebug();
+
+		Hooks.resetOnEachOperator();
+		Hooks.resetOnLastOperator();
+
+		Hooks.resetOnErrorDropped();
+		Hooks.resetOnNextDropped();
+
+		Hooks.resetOnNextError();
+		Hooks.resetOnOperatorError();
+
+		Schedulers.resetOnHandleError();
+		Schedulers.resetFactory();
+		Schedulers.resetOnScheduleHooks();
+
+		// TODO capture non-default schedulers and shutdown them
+	}
+
+	@Override
+	public void executionFinished(TestIdentifier testIdentifier, TestExecutionResult testExecutionResult) {
+		reset();
```

Comment:
Do you want to invoke `reset()` for all test identifiers?

FYI: as currently implemented, `reset()` would get invoked after every test method and after every test class.

So, if you just want to do this for test methods, you would need to check if the `TestIdentifier` represents a container or a test.

Reply:
Since `reset` method is very lightweight, there is no harm in calling it for both test method & class.
We had a concern that some classes may install class-level hooks, but it seems that we don't have any.

Reply:
OK. Thanks for the clarification.